### PR TITLE
Add OrderStatistics related classes and variables

### PR DESCRIPTION
**Pull request by adamzzq**
* StatsLogic: separate interface from main to handle restaurant performance related commands
* StatsCommand: implementations of view all kinds of performance data
* Add new CommandTypes to facilitate StatsLogic
* Refactor stats-related classes into packages

### CODE REVIEW

File: src/main/java/logic/StatsLogic.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                break;
+            case QUIT:
+                hasQuit = true;
+                break;
+            case HELP:
+                StatsHelpCommand.execute();
+                break;
+            default:
+                System.out.println("Invalid command. Please try again.");
+                break;
```
Comment (Xb990219) [inline src/main/java/logic/StatsLogic.java:61]: Very clear format

File: src/main/java/command/stats/StatsProfitCommand.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+public class StatsProfitCommand implements StatsCommand {
+    /**
+     * Prints the profit in the order statistics
+     *
+     * @param stats OrderStatistics object containing the statistics
+     * @param inputText the input text that was entered by the user
+     */
+    public static void execute(OrderStatistics stats, String inputText) {
+        String[] indexString = Parser.splitInput(Parser.analyzeInput(inputText), inputText);
+        double cost = Double.parseDouble(indexString[0]);
```
Comment (webtjs) [inline src/main/java/command/stats/StatsProfitCommand.java:15]: Does this need to be in a try-catch block in case parseDouble() throws an exception?

**Review by webtjs — COMMENTED** [review]
Great work, code is neat and easy to understand.

File: src/main/java/command/stats/StatsHelpCommand.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    /**
+     * Prints the help message for the stats commands
+     */
+    public static void execute() {
+        System.out.println("Available commands:");
+        System.out.println("\tbestselling - View the best selling item(s)");
+        System.out.println("\ttotal orders - View the total number of orders");
+        System.out.println("\trevenue -gross - View the gross revenue");
+        System.out.println("\trevenue -net - View the net revenue");
+        System.out.println("\tview profit -cost <cost> - View the profit based on the cost");
```
Comment (Zhengwinter) [inline src/main/java/command/stats/StatsHelpCommand.java:13]: What does profit based on the cost mean? Doesn't gross revenue reflect revenue without costs, which is the profit?

Comment (Zhengwinter) [reply src/main/java/command/stats/StatsHelpCommand.java:13]: Upon some research I discovered that net revenue is not the same as profit, but you could clarify the command either in the user guide or above so it is clearer for users without the necessary knowledge

**Review by Zhengwinter — COMMENTED** [review]
Generally well written and sound, but it'll be great to clarify gross revenue, net revenue and profit either within the code or in the user guide so the user does not confuse between them

File: src/main/java/command/stats/StatsHelpCommand.java (hunk 3 above)
Comment (adamzzq) [reply src/main/java/command/stats/StatsHelpCommand.java:13]: there are more detailed explanations of profit and revenue in the UG; can add on if they are not precise enough

**Pull request merged**